spring-ws-core/src/main/java/org/springframework/ws/soap/addressing/version/Addressing200408.java

```diff
@@ -52,7 +52,9 @@ public boolean hasRequiredProperties(MessageAddressingProperties map) {
 		if (map.getAction() == null) {
 			return false;
 		}
-		if (map.getReplyTo() != null || map.getFaultTo() != null) {
+		
+		if ((map.getReplyTo() != null && !hasAnonymousAddress(map.getReplyTo()))  
+				|| (map.getFaultTo() != null && !hasAnonymousAddress(map.getFaultTo()))){  
```

Comment:
See previous comment. If there is a shared function between them, entertain creating a share utility class class.